src/main/java/sparksoniq/jsoniq/runtime/iterator/functions/sequences/general/TreatAsClosure.java

```diff
@@ -27,6 +27,12 @@ public Boolean call(Item input) throws Exception {
                         + sequenceType.getArity().getSymbol(),
                     metadata
             );
+        if (input != null && sequenceType.isEmptySequence())
```

Comment:
Can input ever be null (here and above)? If the input RDD is empty, call() will not be called at all...

Reply:
That's right, there is no real need for this extra check. Thank you

Reply:
Thanks a lot. I cannot see any pushed commits removing the "input != null" part though (the branch looks unchanged). Is it an overlook?

Reply:
No it's not an overlook, I was waiting to reach an agreement on the "_currentResult_" discussion to push both changes.